subsys/storage/flash_map/flash_map.c

```diff
@@ -176,11 +172,18 @@ static bool get_sectors_cb(const struct flash_pages_info *info, void *datav)
 	return true;
 }
 
-int flash_area_get_sectors(int idx, uint32_t *cnt, struct flash_sector *ret)
+int flash_area_get_sectors(const uintptr_t id, uint32_t *cnt, struct flash_sector *ret)
 {
 	struct layout_data data;
+	const struct flash_area *fa;
+
+	int rc = flash_area_open(id, &fa);
```

Comment:
Aren't you technically supposed to close these areas when you're done with them?

I know that the open/close thing is yet another mynewt legacy and zephyr has never done anything on close, but when I was more involved here, the idea was that close might do something sometime, so it had to be called each time.

Otherwise you might as well just have something like this within zephyr:

```
/**
 * @param fa_id flash area ID
 * @return a pointer to the flash_area, or NULL if it does not exist
 */
struct flash_area *flash_area_from_id(int fa_id);
```